### PR TITLE
Fix Dynamo benchmark model state isolation

### LINKED ISSUE #145773: Dynamo performance test benchmarks reuse same model object between eager and compiled
### 🐛 Describe the bug

See `model`:
https://github.com/pytorch/pytorch/blob/64cd81712ddf867d1c4dc46ba4554d40d6d7d610/benchmarks/dynamo/common.py#L3431-L3470

Later, it is still used by the `speedup_experiment`. This can cause issues with stateful models like convit_base: https://github.com/huggingface/pytorch-image-models/blob/d81da93c1640a504977b0ee494791e5c634ec63c/timm/models/convit.py#L67-L70. 

### Versions

main

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185929

Dynamo performance benchmarks were intended to compare eager and compiled
runs without letting one mode mutate the other mode's model state. The
performance runner deep-copied the loaded model once, but then reused that
single copy for eager warmup, compiled warmup, and the final interleaved
speedup_experiment measurement. Stateful models and wrappers could therefore
let eager timing mutate state before compiled timing saw the model.

The timing helpers also requested and retained outputs that were unused for
performance reporting. For CUDAGraph-backed compiled inference, keeping those
outputs alive across later replays can trip the stale-output overwrite check.

Create separate eager and compiled model copies for performance measurements,
and use separate optimizers when benchmarking training paths. Pass the eager
model and scoped optimizer contexts into speedup_experiment so the interleaved
measurement still compares the two modes fairly without sharing module state.
Drop unused timing outputs immediately instead of keeping references alive.

HF LLM benchmarks route through model.generate, so make that dispatch use the
model argument supplied by the runner rather than a closure over the originally
loaded model. This preserves the same eager/compiled model separation for
LLM generation benchmarks.

The previous linked PR skipped the failing CUDAGraph models in performance
benchmarks. This change fixes the benchmark harness root cause instead of
adding more model-specific skip entries.

Benchmark Results:
- Command: standalone Python comparison of pre-fix timed behavior vs current
  benchmarks.dynamo.common.timed with return_result=False, 20,000 no-op-output
  iterations per rep, 5 reps.
- Old seconds: 1.806908, 1.790106, 1.783095, 1.779866, 1.758974.
- New seconds: 1.894045, 1.884448, 1.877848, 1.881107, 1.852486.
- Medians: old 1.783095s, new 1.881107s, new/old 1.054967x.
  This is a worst-case no-op helper loop; real benchmark model runtimes should
  dominate this small Python reference-clearing overhead. The fix does not add
  cudagraph step markers or synchronization inside the measured loop.

Fixes #145773
Generated by my agent

Test Plan:
- python -m py_compile benchmarks/dynamo/common.py benchmarks/dynamo/huggingface.py benchmarks/dynamo/test.py
- python -m unittest benchmarks.dynamo.test.TestDynamoBenchmark.test_timed_does_not_retain_outputs_when_result_unused benchmarks.dynamo.test.TestDynamoBenchmark.test_speedup_experiment_uses_separate_eager_model benchmarks.dynamo.test.TestDynamoBenchmark.test_call_model_generate_uses_passed_model
- lintrunner -a benchmarks/dynamo/common.py benchmarks/dynamo/huggingface.py benchmarks/dynamo/test.py
- CUDA cudagraph sanity check with torch.compile(..., backend='inductor') and timed(..., return_result=False), passed locally.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says this PR is being closed as part of a cleanup of open agentic PRs to focus review bandwidth on a smaller set; the linked issue remains open.
